e2e/clone_test.go

```diff
@@ -0,0 +1,229 @@
+
+	//go:embed testdata/cloning/clone-pvc-template.yaml
+	thinPvcCloneTemplateYAML string
+)
+
+const (
+	thinClonePVCName = "thin-clone"
+	thinClonePodName = "thin-clone-pod"
+)
+
+func testPVCClone() {
+	nsCloneTest := "clone-test" + randomString(10)
+	BeforeEach(func() {
+		createNamespace(nsCloneTest)
+	})
+	AfterEach(func() {
+		if !CurrentGinkgoTestDescription().Failed {
+			kubectl("delete", "namespaces/"+nsCloneTest)
+		}
+	})
+
+	It("should create a PVC Clone", func() {
+		By("deploying Pod with PVC")
+
+		nodeName := "topolvm-e2e-worker"
+		if isDaemonsetLvmdEnvSet() {
+			nodeName = getDaemonsetLvmdNodeName()
+		}
+		var volumeName string
+		thinPvcYAML := []byte(fmt.Sprintf(thinPVCTemplateYAML, volName, pvcSize))
+		stdout, stderr, err := kubectlWithInput(thinPvcYAML, "apply", "-n", nsCloneTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		thinPodYAML := []byte(fmt.Sprintf(thinPodTemplateYAML, "thinpod", volName, nodeName))
+		stdout, stderr, err = kubectlWithInput(thinPodYAML, "apply", "-n", nsCloneTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		By("confirming if the source PVC and application")
+		Eventually(func() error {
+			stdout, stderr, err = kubectl("get", "pvc", volName, "-n", nsCloneTest)
+			if err != nil {
+				return fmt.Errorf("failed to create PVC. stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			}
+
+			stdout, stderr, err = kubectl("get", "pods", "thinpod", "-n", nsCloneTest)
+			if err != nil {
+				return fmt.Errorf("failed to create Pod. stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			}
+
+			return nil
+		}).Should(Succeed())
+
+		By("writing a file on mountpath")
+		writePath := "/test1/bootstrap.log"
+		Eventually(func() error {
+			stdout, stderr, err = kubectl("exec", "-n", nsCloneTest, "thinpod", "--", "cp", "/var/log/bootstrap.log", writePath)
+			return err
+		}).Should(Succeed())
+
+		stdout, stderr, err = kubectl("exec", "-n", nsCloneTest, "thinpod", "--", "sync")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+		stdout, stderr, err = kubectl("exec", "-n", nsCloneTest, "thinpod", "--", "cat", writePath)
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+		Expect(strings.TrimSpace(string(stdout))).ShouldNot(BeEmpty())
+
+		thinPVCCloneYAML := []byte(fmt.Sprintf(thinPvcCloneTemplateYAML, thinClonePVCName, "thinvol", pvcSize))
+		stdout, stderr, err = kubectlWithInput(thinPVCCloneYAML, "apply", "-n", nsCloneTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		thinPodCloneYAML := []byte(fmt.Sprintf(thinPodCloneTemplateYAML, thinClonePodName, thinClonePVCName, nodeName))
+		stdout, stderr, err = kubectlWithInput(thinPodCloneYAML, "apply", "-n", nsCloneTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
```

Comment:
Please confirm that the clone PVC was created and is bound.

Reply:
Added the check :+1: